azure-pipelines.yml

```diff
@@ -15,26 +15,30 @@ variables:
 steps:
 - script: |
     GIT_SHORT_SHA=$(echo $(Build.SourceVersion) | cut -c 1-7)
-    tag=$(dockerHubUsername)/$(imageName):$GIT_SHORT_SHA
+    docker_path=$(dockerHubUsername)/$(imageName)
     echo "##vso[build.updatebuildnumber]$GIT_SHORT_SHA"
-    echo "##vso[task.setvariable variable=tag;]$tag"
+    echo "##vso[task.setvariable variable=docker_path;]$docker_path"
   displayName: 'Set version number'
 
 - script: |
-    docker build -f Dockerfile -t $(tag) .
-  displayName: 'Build image'
+    docker-compose build
+    docker-compose up --no-build -d
+    docker-compose exec -T web /bin/sh -c "./wait-for-command.sh -c 'nc -z db 5432' -s 0 -t 20"
+    docker-compose exec -T web /bin/sh -c "bundle exec rails db:setup"
+  displayName: 'Build & setup'
 
 - script: |
-    docker run "$(tag)" /bin/sh -c "bundle exec rake spec SPEC_OPTS='--format RspecJunitFormatter'" | sed -e 1d >> rspec-results.xml
+    docker-compose exec -T web /bin/sh -c 'bundle exec rake spec SPEC_OPTS="--format RspecJunitFormatter --out ./rspec-results.xml"'
   displayName: 'Execute tests'
 
 - script: |
-    docker run "$(tag)" /bin/sh -c "bundle exec govuk-lint-ruby app config db lib spec --format clang"
+    docker-compose exec -T web /bin/sh -c "bundle exec govuk-lint-ruby app config db lib spec --format clang"
   displayName: 'Execute linters'
 
 - script: |
     echo $password | docker login --username $(dockerHubUsername) --password-stdin
```

Comment:
Echoing password? Are you sure?

Reply:
https://docs.docker.com/engine/reference/commandline/login/

See **Provide a password using STDIN**